components/dashboards-web-component/src/utils/DashboardUtils.jsx

```diff
@@ -100,4 +101,76 @@ export default class DashboardUtils {
         return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
             s4() + '-' + s4() + s4() + s4();
     };
+
+    /**
+     * helper function to traverse the content of a page and find widgets to extract their options to a
+     * given map
+     * @param page : the page to be traversed
+     * @param OptionsMap : the map to extract option configurations
+     * */
+    static searchForWidgetsInAPage(page, OptionsMap) {
+        let obj = page;
```

Comment:
Why assign `page` to `obj`? Also `obj` is not a good name for a variable.

Reply:
I'll change obj to objectTobeTraversed, we start traversing a page and traverse through the child objects or content objects of  a page

Reply:
changed with 910980c413d9aea940c47a751f949593c1fc5ec0